### PR TITLE
circleci: Move e2e-mixer back to per-PR runlist

### PR DESCRIPTION
**Pull request by cmluciano**
Signed-off-by: Christopher M. Luciano <cmluciano@us.ibm.com>

fixes: #4023

### CODE REVIEW

**Comment by douglas-reid** [comment]
So, something I noticed:

```
mixer_test.go:278: Could not get metrics from prometheus: Get http://localhost:9090/api/v1/query?query=istio_request_count%7Bdestination_service%3D%22productpage.istio-system.svc.cluster.local%22%7D&time=2018-03-09T17%3A38%3A47.556255324Z: dial tcp 127.0.0.1:9090: getsockopt: connection refused
```

I think what is happening is that the `port-forward` command for `prometheus` is being run before the pod for the deployment is ready. That command is then failing (but not returning an error), causing all subsequent lookups to fail.

I suggest adding the following to the `mixer_test.go` code:
```
func (p *promProxy) Setup() error {

// BEGIN ADDED CODE

	if !util.CheckPodsRunning(tc.Kube.Namespace) {
		return errors.New("could not establish prometheus proxy: pods not running")
	}

// END ADDED CODE
...
}
```

**Comment by cmluciano** [comment]
Cool. After rebasing all the fixes I think I'm in a steady state now. Will try this out locally to see if I can spot it prior to changing the test, but that fix sounds reasonable to me.

File: .circleci/config.yml [hunk 1]
```diff
@@ -650,3 +647,6 @@ workflows:
       - e2e-simple:
           requires:
             - build
+      - e2e-mixer:
+          requires:
+            - build
```
Comment (ldemailly) [inline .circleci/config.yml]: add the missing new line (and set your editor to not do it again)

Comment (cmluciano) [reply .circleci/config.yml]: will do

File: tests/e2e/tests/mixer/mixer_test.go [hunk 2]
```diff
@@ -211,12 +211,15 @@ func (p *promProxy) portForward(labelSelector string, localPort string, remotePo
 func (p *promProxy) Setup() error {
 	var err error
 
+	if !util.CheckPodsRunning(tc.Kube.Namespace) {
+		return errors.New("could not establish prometheus proxy: pods not running")
+	}
 	if err = p.portForward("app=prometheus", prometheusPort, prometheusPort); err != nil {
-		return err
+	return err
```
Comment (douglas-reid) [inline tests/e2e/tests/mixer/mixer_test.go]: curious: what happened with the formatting here ?

**Comment by cmluciano** [comment]
@douglas-reid sry about that, I just added the newline in the circleci file. This should be all clear now.

**Comment by cmluciano** [comment]
@douglas-reid This should be good to go now

**Pull request merged**